src/protocol/clock/timeextent.rs

```
@@ -0,0 +1,185 @@
+        Clock::now()
+            .as_nanos()
+            .checked_div((*increment).as_nanos())
+            .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                Err(error) => Err(error),
+                Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+            })
+    }
+
+    fn now_add(increment: &TimeExtentBase, add_time: &Duration) -> Option<Result<TimeExtent, TryFromIntError>> {
+        match Clock::add(add_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(increment.as_nanos())
+                    .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                        Err(error) => Err(error),
+                        Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+                    })
+            }
+        }
+    }
+    fn now_sub(increment: &TimeExtentBase, sub_time: &Duration) -> Option<Result<TimeExtent, TryFromIntError>> {
+        match Clock::sub(sub_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(increment.as_nanos())
+                    .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                        Err(error) => Err(error),
+                        Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+                    })
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TimeExtentMaker<const CLOCK_TYPE: usize> {}
+
+pub type WorkingClockTimeExtentMaker = TimeExtentMaker<{ ClockType::WorkingClock as usize }>;
+pub type StoppedClockTimeExtentMaker = TimeExtentMaker<{ ClockType::StoppedClock as usize }>;
+
+impl MakeTimeExtent<WorkingClock> for WorkingClockTimeExtentMaker {}
+impl MakeTimeExtent<StoppedClock> for StoppedClockTimeExtentMaker {}
+
+#[cfg(not(test))]
+pub type DefaultClockTimeExtentMaker = WorkingClockTimeExtentMaker;
```

@da2ce7 can we just use `DefaultTimeExtentMaker`? We can hide the implementation detail. Especially because we do not have any other implementation. The name is confusing for me probably because I'm not an English native speaker. It could be:

- The `Default` `ClockTimeExtentMaker`
- or the `TimeExtentMaker` which uses the `DefaultClock`

Following your conventions, it should be `DefaultDeafultClockTimeEntentMaker` because it's the default implementation of the TimeExtentMaker using the DefaultClock.

I would use: 

- `DefaultTimeExtentMaker` (removing information about where the time comes from)
- or `DefaultTimeExtentMakerUsingDefaultClock`

---

I think that `DefaultTimeExtentMaker` is better. :)

---

Done :)

---

Done :)